### PR TITLE
Prevent crash by validating Filter Menu data

### LINKED ISSUE #1193: Crash in onFilterFocusChange()
## Software Versions

- Jellyfin Server Version:  N/A
- Roku Client Version:  v1.6.4

## Describe the bug
App crash found in `ItemGridOptions.brs` [line 56](https://github.com/jellyfin/jellyfin-roku/blob/v1.6.4/components/ItemGrid/ItemGridOptions.brs#L56) via roku.com crash log

## Logs
```
'Dot' Operator attempted with invalid BrightScript Component or interface reference. (runtime error &hec) in pkg:/components/ItemGrid/ItemGridOptions.brs(56) 
Backtrace: 
#0  Function onfilterfocuschange() As Void file/line: pkg:/components/ItemGrid/ItemGridOptions.brs(56) 
Local Variables: 
global           Interface:ifGlobal m                roAssociativeArray refcnt=3 count:19
```

## Screenshots
app-crashes

### PR DESCRIPTION
**Pull request by cewert**
## Changes
- Validate filter menu data before using

## Issues
Fixes #1193

fixes v1.6.4 crash log line 56

### CODE REVIEW

**Comment by 1hitsong** [comment]
Similar logic that was causing the error in the log also appears in onKeyEvent()

Line 269 and 323 access the same m.filtermenu variables, but we're not confirming they're valid before using them.

image
image

**Comment by cewert** [comment]
`m.filterMenu` is also being using three more  times lower in the onKeyEvent() function. I don't see any crashes in the Roku crash logs for any of these lines. 

more-filtermenu

It looks like we could call the new `isFilterMenuDataValid` function immediately if key = right or OK then return false if the function returns false. Does that solution work for you?

**Comment by cewert** [comment]
@1hitsong can you take another look at this, please? I think it's ready to merge as long as my last commit works for you.

File: components/ItemGrid/ItemGridOptions.brs [hunk 1]
```diff
@@ -53,6 +53,11 @@ sub hideChecklist()
 end sub
 
 sub onFilterFocusChange()
+    if isFilterMenuDataValid() = false
```
Comment (1hitsong) [inline components/ItemGrid/ItemGridOptions.brs]:
```suggestion
    if not isFilterMenuDataValid()
```

File: components/ItemGrid/ItemGridOptions.brs [hunk 2]
```diff
@@ -267,6 +285,8 @@ function onKeyEvent(key as string, press as boolean) as boolean
             return true
         end if
     else if key = "OK"
+        if isFilterMenuDataValid() = false then return false
```
Comment (1hitsong) [inline components/ItemGrid/ItemGridOptions.brs]: By checking at the top for OK presses, it means the View and Sort options also won't work if the filter data is bad. I think we should isolate it so we only check the filter data is valid and return false if the user is attempting to view the filter options. That way we don't impact the other options.

**Review by 1hitsong — CHANGES_REQUESTED** [review]

**Review by 1hitsong — APPROVED** [review]

**Pull request merged**